pkg/tun/tun_linux.go

```diff
@@ -0,0 +1,146 @@
+// tun_linux.go: Open a Tunnel (L3 virtual interface) using the Universal TUN/TAP device driver.
+package tun
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"unsafe"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/datawire/dlib/dexec"
+	"github.com/telepresenceio/telepresence/v2/pkg/tun/buffer"
+)
+
+const devicePath = "/dev/net/tun"
+
+type Device struct {
+	*os.File
+	name  string
+	index uint32
+}
+
+// OpenTun creates a new TUN device and ensures that it is up and running.
+func OpenTun() (*Device, error) {
+	// https://www.kernel.org/doc/Documentation/networking/tuntap.txt
+
+	fd, err := unix.Open(devicePath, unix.O_RDWR, 0)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open TUN device %s: %v", devicePath, err)
+	}
+
+	var flagsRequest struct {
+		name  [unix.IFNAMSIZ]byte
+		flags int16
+	}
+	copy(flagsRequest.name[:], "tel%d")
+	flagsRequest.flags = unix.IFF_TUN | unix.IFF_NO_PI
```

Comment:
Without `unix.IFF_NO_PI`, it would have 4-byte prefix indicating some flags, and indicating the L3 proto.  We clearly don't need that, but having the 4 byte prefix might be nice, since it would mean that we could use the same `buffer` code on GOOS=darwin and GOOS=linux.  Sure, the 4 byte-prefix doesn't get parsed the same between the two platforms, but having the offset be the same might be nice.

Reply:
Parsing is half the problem. How would we set those four bytes when we write stuff back to the TUN-device? I assume they need to contain correct flags, etc.